testrunner/src/mill/testrunner/TestRunnerUtils.scala

```diff
@@ -347,4 +361,26 @@ import java.io.PrintStream
       filters.exists(f => f(name))
     }
   }
+
+  @internal private[TestRunnerUtils] final class TestEventSummary(outputFileOpt: Option[os.Path])
+      extends AtomicBoolean(true) {
+    def record(event: Event): Unit = {
+      event.status() match {
+        case Status.Error => set(false)
+        case Status.Failure => set(false)
+        case _ => val _ = compareAndSet(true, true) // consider success as a default
+      }
+    }
+
+    def summary(): Boolean = {
+      val isSuccess = get()
+      outputFileOpt.foreach { outputFile =>
+        val (success, failure) = upickle.default.read[(Long, Long)](os.read.stream(outputFile))
+        val (newSuccess, newFailure) =
+          if (isSuccess) (success + 1, failure) else (success, failure + 1)
```

Comment:
This workflow to both read and write to the file on disk is unnecessarily complicated. Let's just keep the success/failure count as in-memory `Long` (or `AtomicLong`) and have the file be write-only purely for the purpose of passing the information to the parent process